### ISSUE TITLE
Hand of Protection while physical cc'd

### ISSUE THREAD

**Issue by mauvas** [TrinityCore/TrinityCore#12101]
Rev: https://github.com/TrinityCore/TrinityCore/commit/b41de6cb38636f17d95810366bbfb593fea8acc7
DB: TDB.335.53
OS: Win7 64

Paladins are able to use Hand of Protection while they're under effects of some CC effects, ex Blind, Stun, Gouge (physical type CC)

I think the problem is in Spell::CheckCasterAuras(), because the core makes you able to cast spell under CC if spell gives inmunity to that CC (hand of protection makes inmune to physical CC) and also has SPELL_ATTR1_DISPEL_AURAS_ON_IMMUNITY. This is fine as long as the paladin is the unit target of the HoP spell, you have some aurastate that shouldn't allow you to cast, but since spell gives inmunity, you can cast.

However, you shouldn't be able to HoP party members while physical CC'd.
For other CC types its working fine (paladin is polymorphed, can't cast HoP on noone)

Check should return false if target of spell is not the caster, not really sure how to fix it thought

**Comment by ariel-** [comment]
Closed by https://github.com/TrinityCore/TrinityCore/commit/625ca6ec1ca524f7435b34408470c9a04b675566

**Comment by joschiwald** [comment]
it's a different issue

**Comment by Killyana** [comment]
This issue seems to be wrong , Hand of Freedom should be usable on a Friendly Target while the Paladin is in Sap and some others CC's.
https://github.com/trinitycore/trinitycore/issues/7206

**Comment by ariel-** [comment]
Hand of Freedom is not Hand of Protection

**Comment by Killyana** [comment]
Yes, may bad, any way I wasn't able to find any source about that.

**Comment by Keader** [comment]
What about:

```diff

diff --git a/src/server/game/Spells/SpellInfo.cpp b/src/server/game/Spells/SpellInfo.cpp
index 58ac6f6..81ebd3b 100644
--- a/src/server/game/Spells/SpellInfo.cpp
+++ b/src/server/game/Spells/SpellInfo.cpp
@@ -2804,6 +2804,13 @@ void SpellInfo::_LoadImmunityInfo()
                 break;
             case 49039: // Lichborne, don't allow normal stuns
                 break;
+            case 1044: // Hand of Freedom
+                _allowedMechanicMask |=
+                    (1 << MECHANIC_STUN) |
+                    (1 << MECHANIC_FREEZE) |
+                    (1 << MECHANIC_SAPPED) |
+                    (1 << MECHANIC_SLEEP);
+                break;
             default:
                 _allowedMechanicMask |= (1 << MECHANIC_STUN);
                 break;
```

**Comment by ariel-** [comment]
I'm not completely sure, is this valid?

**Comment by CraftedRO** [comment]
if this video I found is real, then the issue looks valid : https://youtu.be/swgUTMmawbE?t=446 (try to set playback speed on 0.25 and then watch hand of protection spell borders on his auctionbar.
